byterun/caml/mlvalues.h

```diff
@@ -186,7 +183,24 @@ bits  63        (64-P) (63-P)        10 9     8 7   0
 /* Pointer to the first field. */
 #define Op_val(x) ((value *) (x))
 /* Fields are numbered from 0. */
+#if CAML_API_VERSION < 405
 #define Field(x, i) (((value *)(x)) [i])           /* Also an l-value. */
+#endif
+
+static inline void caml_read_field (value obj, mlsize_t f, value* ret)
+{
+  *ret = Op_val(obj)[f];
+}
+
+static inline value Field_imm (value obj, mlsize_t f)
```

Comment:
I had the same reading confusion at first. (I also wonder whether `Immut_field` reads better.)